edgelet/edgelet-http/src/route/mod.rs

```diff
@@ -151,18 +155,38 @@ where
     type Future = Box<Future<Item = Response<Self::ResBody>, Error = Self::Error> + Send>;
 
     fn call(&mut self, req: Request<Body>) -> Self::Future {
-        let method = req.method().clone();
-        let path = req.uri().path().to_owned();
-        match self.inner.recognize(&method, &path) {
-            Ok((handler, params)) => {
-                Box::new(handler.handle(req, params).map_err(|err| err.compat()))
+        let api_version = {
+            let query = req.uri().query();
+            query.and_then(|query| {
+                let mut query = parse_query(query.as_bytes());
+                let (_, api_version) = query.find(|&(ref key, _)| key == "api-version")?;
+
+                let api_version = api_version.into_owned().parse::<Version>();
+                match api_version {
+                    Ok(api_version) => Some(api_version),
+                    Err(_) => None,
+                }
+            })
+        };
+
+        match api_version {
+            Some(ref api_version) => {
+                let method = req.method().clone();
```

Comment:
Don't need to clone `method` since you only use it as a borrow.

Reply:
If method and path are not cloned, there is a compilation error at line 174 cannot move out of `req` because it is borrowed 

---
In reply to: [243431111](https://github.com/Azure/iotedge/pull/656#discussion_r243431111) [](ancestors = 243431111)